torch/csrc/utils/python_dispatch.cpp

```diff
@@ -122,6 +190,12 @@ void initDispatchBindings(PyObject* module) {
       );
       return self;
     }, "", py::arg("name"), py::arg("dispatch") = "", py::arg("debug") = "")
+    .def("impl", [](py::object self, const char* name, const char* dispatch, py::object func) {
+      self.cast<torch::Library&>().impl(
+        name,
+        dispatch_str(dispatch, CppFunction::makeFromBoxedFunctor(std::make_unique<PythonKernelHolder>(std::move(func))))
```

Comment:
I think we should set the debug string on the CppFunction here, so that C++ can give a better error message about where a library was defined at. I suggest capturing one frame of the backtrace on Library construction, extracting out the file and the line, and then passing that in as the debug info here.